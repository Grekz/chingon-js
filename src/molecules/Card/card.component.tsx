import styled from '@emotion/styled'
import * as React from 'react'
import Text from '../../atoms/Text'

interface IImageProps {
  image?: string
  altImage?: string
}

interface ICardWrapperProps {
  width?: string
  backgroundColor?: string
}

interface IBodyProps {
  header?: React.ReactNode
  body?: React.ReactNode
  footer?: React.ReactNode
  children?: React.ReactNode
}

interface ICardProps extends IImageProps, IBodyProps, ICardWrapperProps {}

const Card = ({
  image,
  altImage,
  header,
  body,
  footer,
  children,
  width = '300px',
}: ICardProps) => (
  <CardWrapper width={width}>
    {image ? <CardImage src={image} alt={altImage} /> : null}
    <div>
      {header ? <Text type="h3">{header}</Text> : null}
      <Text type="span">{children || body}</Text>
      {footer ? <>{footer}</> : null}
    </div>
  </CardWrapper>
)

const CardWrapper = styled.div`
  width: ${({ width }: ICardWrapperProps) => (width ? width : '300px')};
  position: relative;
  display: flex;
  flex-direction: column;
  min-width: 0;
  word-wrap: break-word;
  background-color: ${({ backgroundColor }: ICardWrapperProps) =>
    backgroundColor ? backgroundColor : '#fff'};
  background-clip: border-box;
  border: 1px solid rgba(0, 0, 0, 0.125);
  border-radius: 0.25rem;
  > div {
    flex: 1 1 auto;
    padding: 1.25rem;
  }
`
const CardImage = styled.img`
  width: 100%;
  border-top-left-radius: calc(0.25rem - 1px);
  border-top-right-radius: calc(0.25rem - 1px);
`
export default Card
