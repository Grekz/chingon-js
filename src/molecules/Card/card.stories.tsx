import * as React from 'react'

// tslint:disable-next-line: no-implicit-dependencies
import { storiesOf } from '@storybook/react'

import Card from '.'

storiesOf('Molecules|Card', module).add('As Default', () => (
  <Card
    image="https://placeholder.pics/svg/300x180"
    altImage="Test placeholder"
    header="Title here"
  >
    Some body content goes here let's see it complete
  </Card>
))
