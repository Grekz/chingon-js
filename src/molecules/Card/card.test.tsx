import { configure, shallow } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import serializer, { matchers } from 'jest-emotion'
import * as React from 'react'
import Card from '.'

configure({ adapter: new Adapter() })
expect.extend(matchers)
expect.addSnapshotSerializer(serializer)

describe('<Card /> rendering', () => {
  it('should render Card', () => {
    const wrapper = shallow(
      <Card
        image="https://placeholder.pics/svg/300x180"
        altImage="Test placeholder"
        header="Title here"
      >
        Some body content goes here let's see it complete
      </Card>
    )
    expect(wrapper).toHaveLength(1)
  })
  it('should match snapshot', () => {
    const tree = shallow(
      <Card
        image="https://placeholder.pics/svg/300x180"
        altImage="Test placeholder"
        header="Title here"
      >
        Some body content goes here let's see it complete
      </Card>
    )
    expect(toJson(tree)).toMatchSnapshot()
  })
})
