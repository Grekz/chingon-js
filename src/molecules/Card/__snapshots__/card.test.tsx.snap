// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> rendering should match snapshot 1`] = `
<Styled(div)
  width="300px"
>
  <Styled(img)
    alt="Test placeholder"
    src="https://placeholder.pics/svg/300x180"
  />
  <div>
    <Text
      color="#000"
      type="h3"
    >
      Title here
    </Text>
    <Text
      color="#000"
      type="span"
    >
      Some body content goes here let's see it complete
    </Text>
  </div>
</Styled(div)>
`;
